File: gosi-kafka-sdk-java/gosi-kafka-sdk-core/src/main/java/com/gosi/kafka/sdk/consumer/GosiRecord.java
package com.gosi.kafka.sdk.consumer;

import org.apache.kafka.common.header.Headers;

/**
 * Encapsulates a consumed Kafka record along with its metadata.
 */
public class GosiRecord<K, V> {
    private final K key;
    private final V value;
    private final String topic;
    private final int partition;
    private final long offset;
    private final String traceId;
    private final Headers headers;
    private final long timestamp;

    public GosiRecord(K key, V value, String topic, int partition, long offset, String traceId, Headers headers, long timestamp) {
        this.key = key;
        this.value = value;
        this.topic = topic;
        this.partition = partition;
        this.offset = offset;
        this.traceId = traceId;
        this.headers = headers;
        this.timestamp = timestamp;
    }

    public K getKey() { return key; }
    public V getValue() { return value; }
    public String getTopic() { return topic; }
    public int getPartition() { return partition; }
    public long getOffset() { return offset; }
    public String getTraceId() { return traceId; }
    public Headers getHeaders() { return headers; }
    public long getTimestamp() { return timestamp; }
}

File: gosi-kafka-sdk-dotnet/Consumer/GosiRecord.cs
namespace Gosi.Kafka.Sdk.Consumer;

using Confluent.Kafka;

public class GosiRecord<TKey, TValue>
{
    public TKey Key { get; }
    public TValue Value { get; }
    public string Topic { get; }
    public int Partition { get; }
    public long Offset { get; }
    public string TraceId { get; }
    public Headers Headers { get; }
    public long Timestamp { get; }

    public GosiRecord(TKey key, TValue value, string topic, int partition, long offset, string traceId, Headers headers, long timestamp)
    {
        Key = key;
        Value = value;
        Topic = topic;
        Partition = partition;
        Offset = offset;
        TraceId = traceId;
        Headers = headers;
        Timestamp = timestamp;
    }
}
